### ISSUE TITLE
Path planner knows if two unexplored systems are connected by a hyperlane

### ISSUE THREAD

**Issue by nickshanks** [endless-sky/endless-sky#10239]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When creating a multi-jump route, and going via two unexplored systems, the path planner knows if they are connected via a hyperlane or not. See screenshot for an example.

### Steps to Reproduce

1. Create a hyperspace route that passes through two adjacent unexplored systems.
2. Sometimes you get the jump drive dashes an sometimes you get the hyperdrive solid line.

### Expected Behavior

You would not know if a hyperlane existed until you explore either of the systems

### Screenshots

<img width="1312" alt="Screenshot 2024-06-19 at 17 14 31" src="https://github.com/endless-sky/endless-sky/assets/206120/b51103d9-0a50-4951-8428-e05418ee311a">

### Link to save file

[Nicholas Shanks.txt](https://github.com/user-attachments/files/15903966/Nicholas.Shanks.txt)

### Operating System

MacOS X

### Game Source

Built from source

### Game Version

51de725e6eac78abf413e8025cc7e8efc3c91c7b

**Comment by nickshanks** [comment]
I agree 100% with the remarks in your footnote. Escorts getting themselves stranded is bubbling to the top of my list of irritating shit in this game. They won't even stop to refuel at the last inhabited planet before a run of uninhabited systems.

**Comment by ThrawnCA** [comment]
A similar issue would affect systems with non-standard jump drive range; the pathfinder knows the real range even if the player shouldn't.
